### PR TITLE
Prometheus set-up

### PR DESCRIPTION
**Pull request by geahaad**
#32 Basic containerised prometheus setup, with scrapers for:
- [docker engine](https://docs.docker.com/config/daemon/prometheus/) (Requires docker to be run in experimental mode)
- [cadvisor](https://github.com/google/cadvisor/)
- [node-exporter](https://prometheus.io/docs/guides/node-exporter/) (Requires native installation on host)

Includes an additional Grafana Dashboard for all metrics collected by node-exporter, based on [1860](https://grafana.com/grafana/dashboards/1860)

### CODE REVIEW

File: setup/dashboard/docker-compose.yml [hunk 1]
```diff
@@ -55,6 +55,40 @@ services:
...
+      - --config.file=/etc/prometheus/prometheus.yml
+    volumes:
+      - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml:ro
+    depends_on:
+      - cadvisor
+    networks:
+      - dashboard-and-logging
+    # https://stackoverflow.com/a/67158212/898
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (PaulWen) [inline setup/dashboard/docker-compose.yml:73]: Nice!! I guess this also simplifies the rest of our configuration as we do not have to switch to `172.17.0.1` for Linux anymore, right?

**Comment by PaulWen** [comment]
Should we update the `Readme.md` to include the installation step of `node_exporter` and name the new dashboard?

File: setup/dashboard/docker-compose.yml (hunk 1 above)
Comment (geahaad) [reply setup/dashboard/docker-compose.yml:73]: Yeah, I myself was surprised that such an elegant workaround for cross-platform exists. Works like a charm on Darwin/Unix, untested on Windows, though.
Exactly, it should help us replace `172.17.0.1`.

**Comment by geahaad** [comment]
> Should we update the `Readme.md` to include the installation step of `node_exporter` and name the new dashboard?

Valid remark - added an installation guide for node-exporter.

**Pull request merged**